src/core/server/http/router/request.ts

```diff
@@ -33,10 +33,12 @@ const requestSymbol = Symbol('request');
  * Request specific route information exposed to a handler.
  * @public
  * */
-export interface KibanaRequestRoute {
+export interface KibanaRequestRoute<Method extends RouteMethod | 'patch' | 'options'> {
   path: string;
-  method: RouteMethod | 'patch' | 'options';
-  options: Required<RouteConfigOptions>;
+  method: Method;
+  options: Method extends 'get'
+    ? Required<Pick<RouteConfigOptions<Method>, 'authRequired' | 'tags'>> // GET requests cannot have payload options
```

Comment:
requires changes when `RouteConfigOptions`  adds new options. shouldn't we Omit `body` instead?

Reply:
You are totally right! Changing it now.